packages/git/src/node/dugite-git.ts

```diff
@@ -272,19 +357,42 @@ export class DugiteGit implements Git {
     }
 
     async diff(repository: Repository, options?: Git.Options.Diff): Promise<GitFileChange[]> {
-        const args = [
-            'diff',
-            '--name-status',
-            '-C',
-            '-M',
-            '-z'
-        ];
+        const args = ['diff', '--name-status', '-C', '-M', '-z'];
         args.push(this.mapRange((options || {}).range));
         if (options && options.uri) {
-            args.push(...['--', Path.relative(FileUri.fsPath(repository.localUri), FileUri.fsPath(options.uri))]);
+            args.push(...['--', Path.relative(this.getFsPath(repository), this.getFsPath(options.uri))]);
         }
         const result = await this.exec(repository, args);
-        return this.nameStatusParser.parse(repository.localUri, result.stdout.trim().split('\0').filter(item => item && item.length > 0));
+        return this.nameStatusParser.parse(repository.localUri, result.stdout.trim());
+    }
+
+    async log(repository: Repository, options?: Git.Options.Log): Promise<CommitWithChanges[]> {
+        // If remaining commits should be calculated by the backend, then run `git rev-list --count ${fromRevision | HEAD~fromRevision}`.
+        // How to use `mailmap` to map authors: https://www.kernel.org/pub/software/scm/git/docs/git-shortlog.html.
+        const args = ['log'];
+        if (options && options.branch) {
+            args.push(options.branch);
+        }
+        const range = this.mapRange((options || {}).range);
+        args.push(...[range, '-C', '-M', '-m']);
+        const maxCount = options && options.maxCount ? options.maxCount : 0;
+        if (Number.isInteger(maxCount) && maxCount > 0) {
+            args.push(...['-n', `${maxCount}`]);
+        }
+        args.push(...['--name-status', '--date=unix', `--format=${this.commitDetailsParser.getFormat(...CommitDetailsParser.DEFAULT_PLACEHOLDERS)}`, '-z', '--']);
+        if (options && options.uri) {
+            const file = Path.relative(this.getFsPath(repository), this.getFsPath(options.uri));
+            args.push(...[file]);
+        }
+        try {
+            const result = await this.exec(repository, args);
+            return this.commitDetailsParser.parse(
+                repository.localUri, result.stdout.trim()
+                    .split(CommitDetailsParser.COMMIT_CHUNK_DELIMITER)
+                    .filter(item => item && item.length > 0));
+        } catch (e) {
```

Comment:
This `catch` is a leftover, I assume. Maybe I left it there. Could you please clean that up?